yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,125 @@
+          - com.ibm.VPD.Error.KeywordNotFound
+          - com.ibm.VPD.Error.EccCheckFailed
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword. Keyword can be read from hardware or from
+          cache.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware Path of the FRU whose keyword needs to be
+                read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Contains data thats required to read IPZ type data or Keyword
+                type data.
+          - name: target
+            type: size
+            description: >
+                Target takes 0/1 to read keyword from cache/hardware
+                respectively.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target
+      errors:
+          - com.ibm.VPD.Error.PathNotFound
+          - com.ibm.VPD.Error.KeywordNotFound
+
+    - name: GetExpandedLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Location code in expanded format.
+      errors:
+          - com.ibm.VPD.Error.PathNotFound
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Invnetory path of the FRU.
+      errors:
+          - com.ibm.VPD.Error.PathNotFound
+
+    - name: CollectFRUVPD
+      description: >
+          Method to perform VPD collection for the given FRU. The FRU should
+          have concurrentlyMaintainable flag set to true in VPD JSON to execute
+          this action.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Invnetory path of the FRU whose VPD needs to be collected.
+      errors:
+          - com.ibm.VPD.Error.PathNotFound
+
+    - name: PerformVPDRecollection
+      description: >
+          Method to check for FRUs replaced and if required, perform
+          recollection of VPD data for them.
+      errors:
+          - com.ibm.VPD.Error.PathNotFound
+
+properties:
+    - name: VPDCollectionStatus
+      type: enum[self.VPDStatus]
+      default: NotStarted
+      description: >
+          Property which provides the status of FRU VPD collection.
+
+enumerations:
+    - name: VPDStatus
+      description: >
+          VPD collection Status
+      values:
```

Comment:
This seems to be a bit confusing here.
In what case do I update partial? 
Is this the status of the whole process of collecting VPD 
In that case first three would be fine.
Or it is status of VPD collection of each FRU?

Reply:
VPD collection is completed but if one or more FRU's collection is failed and remaining FRU's collection is successful, then we will update status as partial

It is for whole process of collecting VPD.

Reply:
I guess Partial is not needed , Do you think otherwise ? do you have any scenario when we would require this ?

Reply:
Again, As I asked before, I would like to first understand what does this status represents?
If it is related to the process of VPD collection then "partial" makes no sense as the process of collection will start, then will be in progress and will eventually end. It does not care if collection fails for any FRU in between. The process itself is done and will mark that collection has been done .
But,
If this reflects the status of collection w.r.t. each FRU then we can discuss.

Reply:
It is related to process of VPD collection

Reply:
If it is related to VPD collection as a whole then we may not need it. As it should not reflect what happened internally each FRU collection.

Reply:
updated

Reply:
The description is not explanatory enough. Please add more details there.